egs/sre16/v2/get_egs_v2.sh

```diff
@@ -0,0 +1,298 @@
+      --max-frames-per-chunk=$max_frames_per_chunk \
+      --kinds-of-length=$kinds_of_length_train \
+      --frames-per-kind=$frames_per_kind \
+      --data-dir=$data/train_set \
+      --output-dir=${data}/train_set/temp || exit 1
+  #sort and uniq
+  for subdir in `dir $data/train_set/temp`; do
+    cat $data/train_set/temp/$subdir/feats.scp.temp | sort | uniq > $data/train_set/temp/$subdir/feats.scp
+    cat $data/train_set/temp/$subdir/utt2spk.temp | sort | uniq > $data/train_set/temp/$subdir/utt2spk
+  done
+  #Generate data_li/spk2utt and utt2label
+  for subdir in `dir $data/train_set/temp`; do
+    utils/utt2spk_to_spk2utt.pl $data/train_set/temp/$subdir/utt2spk > $data/train_set/temp/$subdir/spk2utt
+    utils/sym2int.pl -f 2 $data/spk2int $data/train_set/temp/$subdir/utt2spk \
+      > $data/train_set/temp/$subdir/utt2int
+  done
+fi
+
+if [ $stage -le 2 ]; then
+  echo "$0: Combine, shuffle and split list for training set"
+  #combine feats.scp and shuffle
+  mkdir -p $data/train_set/temp/combine
+  data_combine=$data/train_set/temp/combine
+
+  cat $data/train_set/temp/data_*/feats.scp > ${data_combine}/feats.scp.bak
+  utils/shuffle_list.pl ${data_combine}/feats.scp.bak > ${data_combine}/feats.scp
+  #combine utt2spk
+  cat $data/train_set/temp/data_*/utt2spk > ${data_combine}/utt2spk
+  #generate spk2utt and utt2label
+  utils/utt2spk_to_spk2utt.pl ${data_combine}/utt2spk > ${data_combine}/spk2utt
+  utils/sym2int.pl -f 2 ${data}/spk2int ${data_combine}/utt2spk \
+    > ${data_combine}/utt2int
+
+  # split feats.scp
+  directories=$(for n in `seq $num_train_archives`; do echo ${data_combine}/split/${n}; done)
+  if ! mkdir -p $directories >&/dev/null; then
+    for n in `seq $nj`; do
+      mkdir -p ${data_combine}/split/${n}
+    done
+  fi
+  feat_scps=$(for n in `seq $num_train_archives`; do echo ${data_combine}/split/${n}/feats.scp; done)
+  utils/split_scp.pl ${data_combine}/feats.scp $feat_scps
+  #According to split{n}/feats.scp, generate utt2spk, spk2utt, utt2label
+  for n in `seq $num_train_archives`; do
+    utils/filter_scp.pl ${data_combine}/split/${n}/feats.scp ${data_combine}/utt2spk \
+      > ${data_combine}/split/${n}/utt2spk
+    utils/utt2spk_to_spk2utt.pl ${data_combine}/split/${n}/utt2spk \
+      > ${data_combine}/split/${n}/spk2utt
+    utils/sym2int.pl -f 2 ${data}/spk2int ${data_combine}/split/${n}/utt2spk \
+      > ${data_combine}/split/${n}/utt2int
+  done
+fi
+
+if [ $stage -le 3 ]; then
+  echo "$0: Dump Egs for training set"
+  if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [[ ! -d $dir/storage ]]; then
+    utils/create_split_dir.pl \
+      /export/b{03,04,05,06}/$USER/kaldi-data/egs/sre16/v2/xvector-$(date +'%m_%d_%H_%M')/storage $dir/storage
+  fi
+  if [ -e $dir/storage ]; then
+    echo "$0: creating data links"
+    utils/create_data_link.pl $(for x in $(seq $num_train_archives); do echo $dir/egs.$x.ark; done)
+  fi
+  #Note: if --num-pdfs options is not supply, you must use global utt2int,
```

Comment:
This note is probably not relevant here. Move it inside nnet3-xvector-get-egs-seg